prow/cmd/deck/static/fuzzy-search.js

```diff
@@ -54,62 +54,76 @@ limitations under the License.
      *  1. +3 score for the matching that occurs near the beginning of the string.
      *  2. +5 score for the matching that is not an alphabetical character.
      *  3. +3 score for the matching that the string character is upper case.
-     *  4. +10 score for the matching that matches the uppercase which is just before a 
+     *  4. +10 score for the matching that matches the uppercase which is just before a
      *  separator.
      * @param {number} i
      * @param {string} str
      * @return {number}
-     */ 
+     */
     this.calcScore = function(i, str) {
       var score = 0;
-      var isNotAlphabetical = function (c) {
-        return (c < 65 || (c > 90 && c < 97) || c > 122);
+      var isAlphabetical = function (c) {
+        return (c > 64 && c < 91) || (c > 96 && c < 123);
       };
       // Bonus if the matching is near the start of the string
       if (i < 3) {
         score += 3;
-      } 
+      }
       // Bonus if the matching is not a alphabetical character
-      if (isNotAlphabetical(str.charCodeAt(i))) {
+      if (!isAlphabetical(str.charCodeAt(i))) {
         score += 5;
       }
       // Bonus if the matching is an UpperCase character
       if (str[i].toUpperCase() === str[i]) {
         score += 3;
       }
-      
+
       // Bonus if matching after a separator
-      var separatorBehind = (i === 0 || isNotAlphabetical(str.charCodeAt(i - 1)));
-      if (separatorBehind) {
+      var separatorBehind = (i === 0 || !isAlphabetical(str.charCodeAt(i - 1)));
+      if (separatorBehind && isAlphabetical(str.charCodeAt(i))) {
         score += 10;
         score += (str[i].toUpperCase() === str[i] ? 5 : 0);
       }
       return score;
     };
-    
+
     /**
      * Get maximum score that a string can get against the pattern.
      * @param {string} pttn
      * @param {string} str
      * @return {number}
      */
     this.getMaxScore = function(pttn, str) {
+      // Rewards perfect match a value of Number.MAX_SAFE_INTEGER
+      if (pttn === str) {
+        return Number.MAX_SAFE_INTEGER;
+      }
+      var i = 0;
+      while (i < Math.min(pttn.length, str.length) && pttn[i] === str[i]) {
+        i++;
+      }
+      // If a string is match at least 5 characters, we consider it a significant match
+      // and give it a base score: Number.MAX_SAFE_INTEGER - len(pattern) and the number
+      // of matches will rank the string.
```

Comment:
With the example of `ci-kubernetes-e2e-kubeadm-gce`, every job that begins with `ci-kubernetes-e2e-` is going to trigger this exit point. Wouldn't we want fuzzy search on the remainder? What's the comparison of a list of results with and without this new rank on such a query?

Reply:
FWIW, I don't think we even needed to change the sort behavior necessarily, we just need the "on enter key pressed" handler to take what's in the text box if we haven't actively selected one of the options.

Reply:
@stevekuznetsov nice point! It should not exit like that. I will treat the streak as a bonus then.
@BenTheElder As it is the fuzzy search, your input may not be in the list of items and we cant take anything you entered. We should take the first item that closest to your input. In your case, your input exist in the list and that should be scored higher than any other item and sit on top of the list so you can select it with an enter.

Reply:
I disagree with this. If I wanted an option in the list, I'd click or arrow down to it, or hover it or.... The text box should respect my input otherwise, in the list or not.

Reply:
It's unexpected and user hostile behavior to take me to something I didn't enter or select.

Reply:
okay, I will change the behavior of the enter